src/common/utility_mac.mm

```diff
@@ -185,53 +185,65 @@
         // An error might occur in the code below, but we cannot report anything, so we just ignore them.
 
         if ([fileManager fileExistsAtPath:plistFile]) {
-            auto maybePlist = readPlistFromFile(plistFile);
-            if (!maybePlist) {
-                // broken plist, overwrite it
-                auto result = writeNewPlistFile(plistFile, fullPath, enable);
-                if (!result) {
-                    qCWarning(lcUtility) << Q_FUNC_INFO << result.error();
-                }
-                return;
-            }
-            NSDictionary *plist = *maybePlist;
-
-            id programValue = plist[@"Program"];
-            if (programValue == nil) {
-                // broken plist, overwrite it
-                auto result = writeNewPlistFile(plistFile, fullPath, enable);
-                if (!result) {
-                    qCWarning(lcUtility) << result.error();
+            if (enable) {
```

Comment:
This flag is called `launch` in the function's docstring. This should be corrected/updated (both signature and docstring).